components/console_cmd_mqtt/console_mqtt.c

```diff
@@ -0,0 +1,445 @@
+            } else {
+                mqtt_cfg.credentials.authentication.key = NULL;
+            }
+        }
+
+        client_handle = esp_mqtt_client_init(&mqtt_cfg);
+        if (client_handle == NULL) {
+            ESP_LOGE(TAG, "ERROR: Client init");
+            return 1;
+        }
+
+        /* The last argument may be used to pass data to the event handler, in this example mqtt_event_handler */
+        esp_mqtt_client_register_event(client_handle, ESP_EVENT_ANY_ID, mqtt_event_handler, NULL);
+        esp_mqtt_client_start(client_handle);
+
+        mqtt_config_free(mqtt_cfg);
+
+    } else if (mqtt_args.disconnect->count > 0) {
+        ESP_LOGD(TAG, "Disconnect command received:");
+
+        if (client_handle == NULL) {
+            ESP_LOGE(TAG, "mqtt client not connected");
+            return 0;
+        }
+
+        if (esp_mqtt_client_stop(client_handle) != ESP_OK) {
+            ESP_LOGE(TAG, "Failed to stop mqtt client task");
+            return 1;
+        }
+
+        client_handle = NULL;
+        ESP_LOGI(TAG, "mqtt client disconnected");
+    }
+
+    return 0;
+}
+
+
+esp_err_t set_mqtt_certs(const uint8_t *client_cert_pem_start_i, const uint8_t *client_cert_pem_end_i,
+                         const uint8_t *client_key_pem_start_i, const uint8_t *client_key_pem_end_i,
+                         const uint8_t *server_cert_pem_start_i, const uint8_t *server_cert_pem_end_i)
+{
+    if (!client_cert_pem_start_i || !client_cert_pem_end_i ||
+            (client_cert_pem_start_i > client_cert_pem_end_i)) {
+        ESP_LOGE(TAG, "Invalid mqtt Client certs(%d)\n", __LINE__);
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    if (client_key_pem_start_i && client_key_pem_end_i &&
+            (client_key_pem_start_i >= client_key_pem_end_i)) {
+        ESP_LOGE(TAG, "Invalid mqtt cert key(%d)\n", __LINE__);
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    if (!server_cert_pem_start_i || !server_cert_pem_end_i ||
+            (server_cert_pem_start_i > server_cert_pem_end_i)) {
+        ESP_LOGE(TAG, "Invalid mqtt certs(%d)\n", __LINE__);
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    client_cert_pem_start = client_cert_pem_start_i;
+    client_cert_pem_end = client_cert_pem_end_i;
+    client_key_pem_start = client_key_pem_start_i;
+    client_key_pem_end = client_key_pem_end_i;
+    server_cert_pem_start = server_cert_pem_start_i;
+    server_cert_pem_end = server_cert_pem_end_i;
+
+    return ESP_OK;
+}
+
+
+static int do_mqtt_sub_cmd(int argc, char **argv)
+{
+    int msg_id;
+    int nerrors = arg_parse(argc, argv, (void **)&mqtt_sub_args);
+    if (nerrors != 0) {
+        arg_print_errors(stderr, mqtt_sub_args.end, argv[0]);
+        return 1;
+    }
+
+    if (client_handle == NULL) {
+        ESP_LOGE(TAG, "mqtt client not connected");
+        return 0;
+    }
+
+    if (mqtt_sub_args.unsubscribe->count > 0) {
+        if (mqtt_sub_args.topic->count <= 0) {
+            ESP_LOGE(TAG, UNSUBSCRIBE_HELP_MSG);
```

Comment:
```suggestion
            ESP_LOGE(TAG, UNSUBSCRIBE_HELP_MSG);
            return 0;
```
otherwise, it would crash on the following line

Reply:
Done